pvlib/modelchain.py

```diff
@@ -1500,6 +1506,24 @@ def prepare_inputs(self, weather):
         --------
         ModelChain.complete_irradiance
         """
+        # transfer albedo from weather to mc.system.arrays if needed
+        if isinstance(weather, pd.DataFrame):  # single weather, many arrays
+            if 'albedo' in weather.columns:
+                for array in self.system.arrays:
+                    if hasattr(array, 'albedo'):
+                        raise ValueError('albedo found in both weather and on'
+                                         ' PVsystem.Array Provide albedo on'
+                                         ' one or on neither, but not both.')
+                    array.albedo = weather['albedo']
+        else:  # multiple weather and arrays
+            for w, array in zip(weather, self.system.arrays):
+                if 'albedo' in w.columns:
+                    if hasattr(array, 'albedo'):
+                        raise ValueError('albedo found in both weather and on'
+                                         ' PVsystem.Array Provide albedo on'
+                                         ' one or on neither, but not both.')
+                    array.albedo = w['albedo']
```

Comment:
Maybe a dumb question:  is there any other part of `ModelChain` that modifies the underlying `PVSystem`/`Array` objects?  If not, that's something we should be sure we're comfortable with.

Reply:
There's not any other transfer of data from `ModelChain` to its `PVSystem/Array`. We had a long discussion in #1387 about whether to put `albedo` in the scope of `PVSystem`, or in `weather`. Both have advantages and disadvantages, and it was clear that eliminating either option had a cost.

I think we're handling both cases here, where the typical use of `PVSystem.Array.albedo` is a constant value, and `weather['albedo']` for time series (such as is delivered by the NSRDB now).

Reply:
I am less thinking about where it makes the most sense to store `albedo` and more thinking about whether mutating the inputs is a good idea.  As an example, modifying the `Array`s like this means that calling `mc.run_model(weather)` twice doesn't work unless you `del` the array albedos after the first ModelChain execution.

Reply:
I'm also concerned about modifying the input object. One alternative is to add albedo=None as a kwarg in Array.get_irradiance. If None, use self.albedo, otherwise pass input to the transposition function.